mmbench/workflow/barrier.py

```diff
@@ -137,21 +141,121 @@ def eval_fn(model, loaders, y_train, y_test, eval_fn=None,
         n_models = len(_models)
         iu = np.array(np.triu_indices(n_models, k=0)).T
         mat = np.zeros((n_models, n_models))
+        points_curve = np.zeros((n_models, n_models, n_coeffs))
         for i1, i2 in iu:
             model1 = _models[i1].to(device).eval()
             model2 = _models[i2].to(device).eval()
             state1 = model1.state_dict()
             state2 = model2.state_dict()
             coeffs, metrics = eval_interpolation(
-                model1, state1, state2, [data_train, data_test], eval_fn,
-                n_coeffs=10, eval_kwargs=kwargs)
-            print(coeffs)
-            print(metrics)
-            mat[i1, i2] = np.trapz(metrics, coeffs)
+                copy.deepcopy(model1), state1, state2, [data_train, data_test],
+                eval_fn, n_coeffs=n_coeffs, eval_kwargs=kwargs)
+            ref = metrics[-1] * coeffs + (1 - coeffs) * metrics[0]
```

Comment:
Create a function with documentation.
slope = (y[-1] - y[0]) / (x[-1] - x[0])
intercept = y[0] - slope * x[0]
ref = slope * x + intercept
ref = [min(y1, y2) for y1, y2 in zip(ref, y)]
A = A(x, y) - A(x, ref)